scanner/src/main/kotlin/PathScanner.kt

```diff
@@ -303,19 +299,16 @@ abstract class PathScanner(
         val storageResult = ScanResultsStorage.storage.add(pkg.id, scanResult)
         val filteredResult = scanResult.filterByIgnorePatterns(scannerConfig.ignorePatterns)
 
-        return when (storageResult) {
-            is Success -> filteredResult
-            is Failure -> {
-                val issue = OrtIssue(
-                    source = ScanResultsStorage.storage.name,
-                    message = storageResult.error,
-                    severity = Severity.WARNING
-                )
-                val issues = scanSummary.issues + issue
-                val summary = scanSummary.copy(issues = issues)
-                filteredResult.copy(summary = summary)
-            }
-        }
+        return storageResult.map { filteredResult }.recover {
+            val issue = OrtIssue(
+                source = ScanResultsStorage.storage.name,
+                message = it.message.orEmpty(),
+                severity = Severity.WARNING
+            )
+            val issues = scanSummary.issues + issue
+            val summary = scanSummary.copy(issues = issues)
+            filteredResult.copy(summary = summary)
+        }.getOrThrow()
```

Comment:
Isn't `getOrThrow()` behaving differently from the existing code?
Wouldn't `getOrElse()` be the correct equivalent instead of `recover` and `getOrThrow()`?

Reply:
You're right, `getOrElse()` clearly fits better here. Will change.